datanode.go

```diff
@@ -60,6 +65,37 @@ type dataNode struct {
 	// Errors will only utilize the first labelCounter they find.  The tree is searched
 	// from leaf to root when looking for populated labelCounters.
 	labelCounter Adder
+
+	// proxies contain all proxies (keyed by proxyID) attached to this context.
+	// Once a proxy has been added to this map, it should be passed down to
+	// all descendant dataNodes.  Therefore, any given child of a data node should
+	// have an equal or greater number of proxies compared to its parent.
+	//
+	// In the hierarchy of data collision priority, proxies have higher priority
+	// than other context data (eg: last descendant's data wins), but lower
+	// priority than any error values.
+	//
+	// Proxies construct their own data tree.  The root of the tree is the first
+	// proxy added to the context.  That proxy will get passed down to all descendants.
+	// If any descendant of the primary dataNode attaches another proxy, that new
+	// proxy becomes a descenant of the currently existing proxy.  This way, when we
+	// add a value to a proxy, we are able to walk its ascendancy tree to add the same
+	// valus to all ancestory proxies, thereby guaranteeing that every aded proxy gets
```

Comment:
```suggestion
	// values to all ancestry proxies, thereby guaranteeing that every added proxy gets
```

Reply:
Wow, three in one sentence? Even an extra vowel? That's a new record!

Reply:
I'm sure I can beat you on this in due time